smartsim/_core/launcher/slurm/slurmCommands.py

```diff
@@ -38,10 +41,8 @@ def sstat(args: t.List[str]) -> t.Tuple[str, str]:
     :type args: List of str
     :returns: Output and error of sstat
     """
-    _sstat = _find_slurm_command("sstat")
-    cmd = [_sstat] + args
-    _, out, error = execute_cmd(cmd)
-    return out, error
+    _, out, err = _execute_slurm_cmd("sstat", args, True)
+    return out, err
 
 
 def sacct(args: t.List[str]) -> t.Tuple[str, str]:
```

Comment:
I'm not sure that love that some of the slurm commands raise and some of them silently fail with no real way to tell which does which without checking the implementation.

Potentially more work than it's worth, but I figured I would throw the idea out there. What would your thoughts be on allowing the caller of `sstat`/`sacct`/etc. decide where or not an error should be exposed by allowing them to set a kw-only `raise_on_err` flag that is passed directly to `_execute_slurm_cmd` and for ALL slurm commands is default `False`?
```py
def sacct(args: t.List[str], *, raise_on_err: bool = False) -> t.Tuple[str, str]:
```
That way at the call site users would need to specify
```py
sacct(["--noheader", ...], raise_on_err=True)
```
just to make it blisteringly clear to other future readers where we want the commands to raise and where we want the commands to silently fail.

Reply:
OK, after a first moment of confusion, I think I followed your suggestion now.

Reply:
Looks fantastic! Thanks for humoring this change even for now even if we decide to rm JIT dependency checking in the near future!